src/test/kotlin/uk/gov/dluhc/printapi/service/PrintResponseProcessingServiceTest.kt

```diff
@@ -0,0 +1,185 @@
+import uk.gov.dluhc.printapi.database.entity.Status.DISPATCHED
+import uk.gov.dluhc.printapi.database.entity.Status.PENDING_ASSIGNMENT_TO_BATCH
+import uk.gov.dluhc.printapi.database.entity.Status.RECEIVED_BY_PRINT_PROVIDER
+import uk.gov.dluhc.printapi.database.entity.Status.SENT_TO_PRINT_PROVIDER
+import uk.gov.dluhc.printapi.database.repository.PrintDetailsRepository
+import uk.gov.dluhc.printapi.printprovider.models.BatchResponse.Status.FAILED
+import uk.gov.dluhc.printapi.printprovider.models.BatchResponse.Status.SUCCESS
+import uk.gov.dluhc.printapi.testsupport.testdata.aValidRequestId
+import uk.gov.dluhc.printapi.testsupport.testdata.entity.buildPrintDetails
+import uk.gov.dluhc.printapi.testsupport.testdata.model.buildBatchResponse
+import uk.gov.dluhc.printapi.testsupport.testdata.model.buildProcessPrintResponseMessage
+import java.time.Clock
+import java.time.Instant
+import java.time.OffsetDateTime
+import java.time.ZoneOffset
+import uk.gov.dluhc.printapi.messaging.models.ProcessPrintResponseMessage.Status as ResponseStatus
+import uk.gov.dluhc.printapi.messaging.models.ProcessPrintResponseMessage.StatusStep as ResponseStatusStep
+
+@ExtendWith(MockitoExtension::class)
+class PrintResponseProcessingServiceTest {
+    private lateinit var service: PrintResponseProcessingService
+
+    @Mock
+    private lateinit var printDetailsRepository: PrintDetailsRepository
+
+    @Mock
+    private lateinit var idFactory: IdFactory
+
+    private val fixedClock = Clock.fixed(Instant.now().minusSeconds(10000), ZoneOffset.UTC)
+
+    @Captor
+    private lateinit var captor: ArgumentCaptor<List<PrintDetails>>
+
+    private val now = OffsetDateTime.now(fixedClock)
+
+    @BeforeEach
+    fun setup() {
+        service = PrintResponseProcessingService(printDetailsRepository, idFactory, fixedClock)
+    }
+
+    @Nested
+    inner class ProcessBatchResponses {
+        @Test
+        fun `should update print details`() {
+            // Given
+            val batchResponse1 = buildBatchResponse(status = FAILED)
+            val batchResponse2 = buildBatchResponse(status = SUCCESS)
+            val batchId1 = batchResponse1.batchId
+            val batchId2 = batchResponse2.batchId
+
+            val printDetails1 = buildPrintDetails(batchId = batchId1, status = SENT_TO_PRINT_PROVIDER)
+            val printDetails2 = buildPrintDetails(batchId = batchId2, status = SENT_TO_PRINT_PROVIDER)
+            val newRequestId = aValidRequestId()
+            given(printDetailsRepository.getAllByStatusAndBatchId(any(), any())).willReturn(
+                listOf(printDetails1),
+                listOf(printDetails2)
+            )
+            given(idFactory.requestId()).willReturn(newRequestId)
+
+            val expectedPrintRequestStatus1 = PrintRequestStatus(
+                status = PENDING_ASSIGNMENT_TO_BATCH,
+                dateCreated = now,
+                eventDateTime = batchResponse1.timestamp,
+                message = null
```

Comment:
This instance will have it's `message` field populated (see my comment on the implementation about processing a `FAILED` batch and setting the `message` property)

Reply:
Thanks, @NathanRussellValtech ,
I will fix this.